### PR TITLE
Fix proteomics experiment processing in atlas

### LINKED ISSUE #68: Debug CyclicGraphException in Snakemake-proteomics
Current baseline proteomics rules are raising an exception.

### PR DESCRIPTION
**Pull request by pmb59**
Fixes to for successful processing of

- baseline proteomics

Reprocessing of the PROT experiments reports a CyclicGraphException. Related to https://github.com/ebi-gene-expression-group/bulk-recalculations/issues/68
```
Cyclic dependency on rule decorate_baseline_proteomics_experiment.
```

Also added some logic adjustments for experiments labeled as 'old' proteomics datasets.

- differential proteomics

### CODE REVIEW

**Review by irisdianauy — COMMENTED** [review]
Changes look good, but I noticed that in rule `processing_steps_newer_datasets` line 128 to 132 is missing the script `gxa_summarize_expression.pl`, and should have been:
```
 perl {workflow.basedir}/bin/gxa_summarize_expression.pl \
    --aggregate-quartiles \
    --configuration {wildcards.accession}-configuration.xml \
    < {wildcards.accession}.tsv.undecorated \
    > {wildcards.accession}.tsv.undecorated.aggregated
```

**Comment by pmb59** [comment]
> Changes look good, but I noticed that in rule `processing_steps_newer_datasets` line 128 to 132 is missing the script `gxa_summarize_expression.pl`, and should have been:
> 
> ```
>  perl {workflow.basedir}/bin/gxa_summarize_expression.pl \
>     --aggregate-quartiles \
>     --configuration {wildcards.accession}-configuration.xml \
>     < {wildcards.accession}.tsv.undecorated \
>     > {wildcards.accession}.tsv.undecorated.aggregated
> ```

You are right @irisdianauy , thanks
Corrected here [a3375dc](https://github.com/ebi-gene-expression-group/bulk-recalculations/pull/71/commits/a3375dc89cd523d8b315ca8d95aaa54f72926f93)

**Review by irisdianauy — APPROVED** [review]
LGTM

File: Snakefile-proteomics [hunk 1]
```diff
@@ -38,11 +38,21 @@ rule check_pride_tsv_files:
...
-        check_tsv_data_file_not_empty_of_data {wildcards.accession}.tsv 3
-
+        if [ {params.is_old} == "False" ]; then
+            # currently check_tsv_data_file_not_empty_of_data does not exit if the files do not exist
+            if [ -e {wildcards.accession}.tsv.undecorated ] && [ -e {wildcards.accession}.tsv.undecorated.aggregated ] && [ -e {wildcards.accession}.tsv ]; then
+                check_tsv_data_file_not_empty_of_data {wildcards.accession}.tsv.undecorated 2
+                check_tsv_data_file_not_empty_of_data {wildcards.accession}.tsv.undecorated.aggregated 2
+                check_tsv_data_file_not_empty_of_data {wildcards.accession}.tsv 3
+            else
+                echo "Starting files do not exist for proteomics experiment {wildcards.accession} "
```
Comment (anilthanki) [inline Snakefile-proteomics:45]: This works well but might good to state which file is missing..

I know I am asking for too much..

File: Snakefile-proteomics [hunk 2]
```diff
@@ -38,11 +38,21 @@ rule check_pride_tsv_files:
...
+                exit 1
+            fi
+        else
+            # these files do not initially exist for older proteomics datasets
+            echo "Skipping check_tsv_data_file_not_empty_of_data for {wildcards.accession} "
```
Comment (anilthanki) [inline Snakefile-proteomics:50]:
```suggestion
            echo "Skipping check_tsv_data_file_not_empty_of_data for {wildcards.accession} as it's a new experiment"
```

**Review by anilthanki — APPROVED** [review]
LGTM

**Pull request merged**